Core/Games/Factorio.cs

```diff
@@ -0,0 +1,398 @@
+        /// </summary>
+        /// <param name="steamPath">Steam Library Path</param>
+        /// <returns>The Factorio path.</returns>
+        private static string FactorioDirectory(string steamPath)
+        {
+            // There are several possibilities for the path under Linux.
+            // Try with the uppercase version.
+            string installPath = Path.Combine(steamPath, "SteamApps", "common", "Factorio");
+
+            if (Directory.Exists(installPath))
+            {
+                return installPath;
+            }
+
+            // Try with the lowercase version.
+            installPath = Path.Combine(steamPath, "steamapps", "common", "Factorio");
+
+            if (Directory.Exists(installPath))
+            {
+                return installPath;
+            }
+            return null;
+        }
+
+        private static readonly ILog log = LogManager.GetLogger(typeof(Factorio));
+    }
+
+    // Thanks, CFAN!
+    public class FactorioInfoJson
+    {
+        [JsonProperty(Required = Required.Always)]
+        public string name;
+
+        [JsonProperty(Required = Required.Always)]
+        public ModuleVersion version;
+
+        public GameVersion factorio_version;
+
+        [JsonProperty(Required = Required.Always)]
+        public string title;
+
+        [JsonProperty(Required = Required.Always)]
+        [JsonConverter(typeof(JsonSingleOrArrayConverter<string>))]
+        public List<string> author;
+
+        public string contact;
+
+        public string homepage;
+
+        public string description;
+
+        public string[] dependencies;
+    }
+
+    public class FactorioAggregatorData
+    {
+        [JsonProperty("x-source")]
+        public string XSource;
+
+        [JsonProperty("factorio-com-id")]
+        public string factorioComId;
+
+        [JsonProperty("factorio-com-source")]
+        public string factorioComSource;
+
+        [JsonProperty("requires-factorio-token")]
+        public string requiresFactorioToken;
+    }
+
+    public class FactorioModule
+    {
+        [JsonProperty(Required = Required.Always)]
+        public FactorioInfoJson modInfo;
+
+        public string[] categories;
+        public string[] tags;
+        public string[] suggests;
+        public string[] recommends;
+        public string[] conflicts;
+
+        [JsonProperty(Required = Required.Always)]
+        public string[] downloadUrls;
+
+        [JsonProperty(Required = Required.AllowNull)]
+        public long downloadSize;
+
+        [JsonProperty(Required = Required.Always)]
+        public string type;
+
+        [JsonProperty(Required = Required.AllowNull)]
+        public DateTime? releasedAt;
+
+        public FactorioAggregatorData aggregatorData;
+
+        public CkanModule ToCkan()
+        {
+            if (downloadUrls.Length < 1)
+            {
+                // No modpacks, thanks
+                return null;
+            }
+            var module = new CkanModule()
```

Comment:
The empty `CkanModule()` constructor isn't supposed to be used outside of tests (and JSON deserialization), according to its doc string:
https://github.com/KSP-CKAN/CKAN/blob/ae3dd4443cf6d693896565aa3bc0aa1ad3e71b41/Core/Types/CkanModule.cs#L269-L279

It's missing health checks, and doesn't populate the `Searchable*` fields.

Should we create a new constructor which takes all the (required) fields as individual parameters instead?

Reply:
Heh, nobody told me that for the DLC stuff, apparently. :grimacing: 

https://github.com/KSP-CKAN/CKAN/blob/ae3dd4443cf6d693896565aa3bc0aa1ad3e71b41/Core/Registry/Registry.cs#L881-L893

On the bright side, we can just call `CalculateSearchables`.

Reply:
Oh, whoops :laughing: 
IMHO we should still at least change that documentation for the empty constructor then.
But preferably we should create a new constructor, which does all the health checks and runs `CalculateSearchables()` at the end, and make the empty constructor private.

Reply:
Agreed, will do.

Also did it for modpacks: :grimacing: 

https://github.com/KSP-CKAN/CKAN/blob/ae3dd4443cf6d693896565aa3bc0aa1ad3e71b41/Core/Registry/RegistryManager.cs#L438-L451

Reply:
> But preferably we should
> - [x] create a new constructor,
> - [x] which does all the health checks
> - [x] and runs `CalculateSearchables()` at the end,
>
> and make the empty constructor private.

Done except for making the empty constructor private, because I don't feel like fighting tests or deserialization code. We just won't use it in those other places anymore.

Reply:
Thank you very much! Feels much better now :)